src/libAtomVM/term.h

```diff
@@ -258,6 +277,36 @@ static inline int term_is_binary(term t)
     return 0;
 }
 
+/**
+ * @brief Checks if a term is a binary
+ *
+ * @details Returns 1 if a term is a binary stored on the heap, otherwise 0.
+ * @param t the term that will be checked.
+ * @return 1 if check succedes, 0 otherwise.
+ */
+static inline int term_is_refc_binary(term t)
+{
+    /* boxed: 10 */
+    if ((t & 0x3) == 0x2) {
+        const term *boxed_value = term_to_const_term_ptr(t);
+        int masked_value = boxed_value[0] & 0x3F;
+        switch (masked_value) {
+            case TERM_BOXED_REFC_BINARY:
+                return 1;
+            default:
+                return 0;
+        }
+    }
+
+    return 0;
+}
+
+static int term_refc_binary_is_const(term t)
+{
+    const term *boxed_value = term_to_const_term_ptr(t);
+    return boxed_value[2] & 0x01;
```

Comment:
What does it mean 0x01 here?

Reply:
boxed_value[2] is a flags field.  The low order bit designates whether the boxed term binary references a const binary or not.  If it's const (as you had it implemented), it is not reference counted.

Reply:
boxed_value[2] is a flags field.  The low order bit designates whether the field is const.  If it is const, the pointer is to the const data, not a `struct RefcBinary`.  See https://github.com/fadushin/AtomVM/blob/refc_binaries/doc/atomvm-memory.md#reference-counted-binaries

Reply:
We should use `#define` so we can avoid hard-coding constants around. Maybe using an enum for flags would be even better.

Reply:
Yeah, +1 to an `enum`

Reply:
Made it an enum